### ISSUE TITLE
PHP warning on unknown database with Doctrine DBAL 2.13.x

### ISSUE THREAD

**Issue by mbrodala** [doctrine/dbal#4645]
### BC Break Report

|    Q        |   A
|------------ | ------
| BC Break    | yes
| Version     | 2.13.1

#### Summary

Connection attempts to a missing database lead to a PHP warning.

#### Previous behaviour

No warning was shown.

#### Current behavior

A PHP warning is issued:
> PHP Warning:  mysqli::real_connect(): (HY000/1049): Unknown database ...

#### How to reproduce

1. Do a connection to a missing database.
2. With Doctrine DBAL 2.11.x no error occurs
3. With Doctrine DBAL 2.13.x a PHP warning occurs

Also see https://github.com/Nimut/testing-framework/issues/136 and https://github.com/TYPO3/testing-framework/issues/233 for more specific examples.

**Comment by morozov** [comment]
Could be solved by backporting c9ce80acd7fc480c6c4585dc285741722919efe3. This will also solve the test failure on non-Linux operating systems:
```
1) Doctrine\Tests\DBAL\Driver\Mysqli\MysqliConnectionTest::testRestoresErrorHandlerOnException
Failed asserting that two strings are identical.
--- Expected
+++ Actual
@@ @@
-'Network is unreachable'
+'Permission denied'

tests/Doctrine/Tests/DBAL/Driver/Mysqli/MysqliConnectionTest.php:56
```

**Comment by bmack** [comment]
@morozov Thank you for this super quick fix 🙇

**Comment by morozov** [comment]
@bmack, you're welcome. Closed via #4647.
